service/src/main/java/greencity/service/ubs/UBSManagementServiceImpl.java

```diff
@@ -1379,7 +1379,13 @@ private Payment buildPaymentEntity(Order order, ManualPaymentRequestDto paymentR
             .orElseThrow(() -> new EntityNotFoundException(EMPLOYEE_NOT_FOUND));
         eventService.save(OrderHistory.ADD_PAYMENT_MANUALLY + paymentRequestDto.getPaymentId(),
             employee.getFirstName() + "  " + employee.getLastName(), order);
-        eventService.save(OrderHistory.ORDER_PAID, OrderHistory.SYSTEM, order);
+        if (order.getOrderPaymentStatus() != null) {
```

Comment:
I see in this method we have that logic
```
 Payment payment = Payment.builder()
            .settlementDate(LocalDate.now().format(DateTimeFormatter.ISO_LOCAL_DATE))
            .amount(paymentRequestDto.getAmount())
            .paymentStatus(PaymentStatus.PAID)
           ...
```

Should we check `order.getOrderPaymentStatus()` before set `paymentStatus`?

Reply:
Yes, I made that verification now